lib/elements/reinforcement/straight_longitudinal.rb

```diff
@@ -0,0 +1,101 @@
+require 'elements/reinforcement/base'
+require 'elements/reinforcement/straight_longitudinal_layer'
+require 'errors/reinforcement/empty_layers'
+
+module Elements
+  module Reinforcement
+    class StraightLongitudinal < Base
+      def initialize(distribution_direction:, above_middle: false)
+        @above_middle = above_middle
+        @layers = []
+        @distribution_direction = distribution_direction
+      end
+
+      def add_layer(start_location:, end_location:, amount_of_rebars:, rebar:)
+        id = @layers.empty? ? 1 : @layers.last.id
+
+        start_location = modify_value_3_of_location(start_location, 0.5 * rebar.diameter)
+        end_location = modify_value_3_of_location(end_location, 0.5 * rebar.diameter)
+
+        @layers << Elements::Reinforcement::StraightLongitudinalLayer.new(
+          start_location:,
+          end_location:,
+          amount_of_rebars:,
+          rebar:,
+          id:,
+          distribution_direction: @distribution_direction
+        )
+
+        @layers.last
+      end
+
+      def change_layer_rebar_configuration(
+        id_of_layer_to_change:,
+        amount_of_new_rebars:,
+        new_rebar:
+      )
+        straight_longitudinal_layer = find(id_of_layer_to_change)
```

Comment:
Let's build `find` as a class method of Layer. That way we centralize both id generation and look up inside the class itself

Reply:
deprecated